spec/models/training_period_spec.rb

```diff
@@ -909,6 +909,37 @@ def will_change_attribute(attribute_to_change:, new_value:)
         expect(relation.includes_values).to include(:school_partnership)
       end
     end
+
+    describe ".closest_to" do
```

Comment:
Might be worth having some common setup for these tests and just changing the date in the individual specs; can we make sure we test:

- training period covering the date
- closest by started on
- closest by finished on
- closest by started on when finished_on is `nil`